backend/src/main/java/ai/giskard/service/ee/LicenseService.java

```diff
@@ -0,0 +1,143 @@
+package ai.giskard.service.ee;
+
+import ai.giskard.config.ApplicationProperties;
+import ai.giskard.service.FileLocationService;
+import ai.giskard.service.GiskardRuntimeException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.util.encoders.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.Base64;
+
+@Service
+@RequiredArgsConstructor
+public class LicenseService {
+
+    private final Logger log = LoggerFactory.getLogger(LicenseService.class);
+
+    private final FileLocationService fileLocationService;
+
+    private String licensePublicKey;
+
+    private License currentLicense;
+
+    @Autowired
+    private ApplicationProperties properties;
+
+    /**
+     * On service init, load the stored license if it exists
+     * Also initialize default license (for now)
+     */
+    @PostConstruct
+    public void init() throws IOException {
+        licensePublicKey = properties.getLicensePublicKey();
+
+        if (currentLicense == null) {
+            if (Files.exists(fileLocationService.licensePath())) {
+                String licenseFile = Files.readString(fileLocationService.licensePath());
+                decodeLicense(licenseFile);
+            } else {
+                // The default license has "active" set to false -> it will go back to the setup page
+                currentLicense = new License();
+            }
+        }
+    }
+
+    public License getCurrentLicense() {
+        synchronized (this) {
+            return currentLicense;
+        }
+    }
+
+    /**
+     * Takes a license file content, parses it, validates that it is a proper license.
+     * If it's a real license, saves it in giskard home and updates available feature flags
+     *
+     * @param licenseFile
+     */
+    public void uploadLicense(String licenseFile) throws IOException {
+        synchronized (this) {
```

Comment:
since the whole method scope is synchronized, it's equivalent to just adding `synchronized` modifier to the method